### PR TITLE
Add hooks in nginx container for certbot

### PR DESCRIPTION
**Pull request by mrjones-plip**
# Description

* `enable_certbot_compatability()` function in [ssl-install.sh](https://github.com/medic/cht-core/blob/8099-nginx-certbot/nginx/ssl-install.sh#L110) -  Creates  the directories to serve `certbot` validation HTTP GET requests
*  `enable_certbot_compatability()` function in [ssl-install.sh](https://github.com/medic/cht-core/blob/8099-nginx-certbot/nginx/ssl-install.sh#L110) - Creates a `/etc/nginx/private/deploy.sh` script to run after `certbot` creates the certs to move them to the right location for `nginx` to see them.
* Note - this PR overrides the use of `AUTO_GENERATE`, as a preemptive re-use of it instead of deprecating it, as we're planning to do.  It also removes `generate_certificate_auto()` from [ssl-install.sh](https://github.com/medic/cht-core/pull/8122/files#diff-1d142cadb0d352501f8ba0fa27b775e113f561125f67e52099cd8414d84cfe4fL67). This seemed less invasive, but I'm open to input!

medic/cht-core#8099

## To test

Assumes FQDN of `deleteme2-certbot-nginx-cht.plip.com` and a CHT SSL docker volume of `cht_cht-ssl`. Both are configurable in the `.env` file for certbot service. Use your own FQDN and verify volume name in `.env`.

Steps 7-11 will be the basis of the new documentation to [be added here.](https://docs.communityhealthtoolkit.org/apps/guides/hosting/4.x/adding-tls-certificates/)

1. provision Ubuntu 22.04 w/ Public IP and FQDN
   *NOTE* - you can not use auto created AWS EC2 names (eg `ec2-35-180-117-38.eu-west-3.compute.amazonaws.com`). You must create a FQDN outside of the  `*.amazonaws.com` domain.
3. install docker
    ```
    curl -fsSL get.docker.com -o get-docker.sh && sh get-docker.sh
    ```
4. Install CHT using compose files from this PR (per `curl` call), create dirs, env file, compose files per [the single node self hosted steps](https://docs.communityhealthtoolkit.org/apps/guides/hosting/4.x/self-hosting-single-node/). Note we set `CERTIFICATE_MODE=AUTO_GENERATE`:
    ```
    sudo apt-get install wamerican
    mkdir -p /home/ubuntu/cht/{compose,certs,upgrade-service,couchdb}
    couchdb_secret=$(shuf -n7 /usr/share/dict/words --random-source=/dev/random | tr '\n' '-' | tr -d "'" | cut -d'-' -f1,2,3,4,5,6,7)
    couchdb_password=$(shuf -n7 /usr/share/dict/words --random-source=/dev/random | tr '\n' '-' | tr -d "'" | cut -d'-' -f1,2,3,4,5,6,7)
    cat > /home/ubuntu/cht/upgrade-service/.env << EOF
    CHT_COMPOSE_PROJECT_NAME=cht
    COUCHDB_SECRET=${couchdb_secret}
    DOCKER_CONFIG_PATH=/home/ubuntu/cht/upgrade-service
    COUCHDB_DATA=/home/ubuntu/cht/couchdb
    CHT_COMPOSE_PATH=/home/ubuntu/cht/compose
    COUCHDB_USER=medic
    COUCHDB_PASSWORD=${couchdb_password}
    COUCHDB_UUID=${uuid}
   CERTIFICATE_MODE=AUTO_GENERATE
    EOF
    cd /home/ubuntu/cht/
    curl -s -o ./compose/cht-core.yml https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8099-nginx-certbot/docker-compose/cht-core.yml
    curl -s -o ./compose/cht-couchdb.yml https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8099-nginx-certbot/docker-compose/cht-couchdb.yml
    curl -s -o ./upgrade-service/docker-compose.yml https://raw.githubusercontent.com/medic/cht-upgrade-service/main/docker-compose.yml
    ```

5. start CHT services:
    ```
    cd /home/ubuntu/cht/upgrade-service
    docker compose up --detach
    ```

7. test `.well-known`  works inside the container CHT nginx container:
   ```shell
   echo "should show 'CERTIFICATE_MODE AUTO_GENERATE':"
   curl http://deleteme2-certbot-nginx-cht.plip.com/.well-known/acme-challenge/index.html
   ```
   
   **NOTE** If this fails, see note below. Do not proceed - it won't work!
   
8. create certbot compose and env file
   ```shell
   mkdir -p /home/ubuntu/cht/certbot
   cd /home/ubuntu/cht/certbot
   cat > docker-compose.yml << EOF
   version: '3.9'
   services:
     certbot:
         container_name: certbot
         hostname: certbot
         image: certbot/certbot
         volumes:
           - ssl-storage:/etc/nginx/private/
           - ssl-storage:/var/log/letsencrypt/
         command: certonly --debug --deploy-hook /etc/nginx/private/deploy.sh --webroot -w /etc/nginx/private/certbot/ --domain \$DOMAIN --non-interactive --key-type rsa --agree-tos --register-unsafely-without-email \$STAGING
   volumes:
     ssl-storage:
         name: \${CHT_SSL_VOLUME}
         external: true
   EOF
   
   cat > .env << EOF
   DOMAIN=deleteme2-certbot-nginx-cht.plip.com
   STAGING=
   CHT_SSL_VOLUME=cht_cht-ssl
   TZ=America/Whitehorse
   EOF
   ```
   
9. generate certs
   ```shell
   cd /home/ubuntu/cht/certbot
   docker compose up
   ```
   
10. reload `nginx`
    ```shell
    docker exec -it cht_nginx_1 nginx -s reload
    ```

11. confirm you have valid certs on domain

12. For a production instance, create a cronjob for once a week to renew certs:
    ```
    #!/bin/sh
    cd /home/ubuntu/cht/certbot
    docker compose up
    ```

### well-known/acme-challenge/index.html failing to show "CERTIFICATE_MODE AUTO_GENERATE"

This is likely because the CI build is not passing so the images on AWS aren't up to date :( The fix is to manually build the `nginx` container and update the image used in the compose file.  Follow these steps

1. stop all services 
   ```
   docker kill $(docker ps --quiet)
   ```
2. checkout cht-core, cd into it, build `nginx` image: 
   ```
   cd
   git clone https://github.com/medic/cht-core.git
   cd cht-core/nginx
   docker build -t nginx-8099 .
   ```
3. replace the image in the compose file with the one you just built:
   ```
   sed -i "s/image:\ public.ecr.aws\/medic\/cht-nginx:4.1.0-8099-nginx-certbot/image: nginx-8099/g" /home/ubuntu/cht/compose/cht-core.yml
   ```
4. go back to step `#5` above and start again ensuring the `curl` call succeeds.

# Code review checklist

- [x] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [x] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [x] Tested: Unit and/or e2e where appropriate
- [x] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs

If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8099-nginx-certbot/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8099-nginx-certbot/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8099-nginx-certbot/docker-compose/cht-couchdb-clustered.yml)
 
# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by mrjones-plip** [comment]
@henokgetachew  - I still need to write the docs for this [over here](https://docs.communityhealthtoolkit.org/apps/guides/hosting/4.x/adding-tls-certificates/), but I'd love validation of my `cron` approach for blind reloads in case there's a new cert as well as an opinion on my removal of `generate_certificate_auto()` and re-use of `AUTO_GENERATE`

much thanks!

File: nginx/ssl-install.sh [hunk 1]
```diff
@@ -130,6 +106,30 @@ select_ssl_certificate_mode(){
...
+}
+
+enable_certbot_compatability(){
+cat > /etc/nginx/private/deploy.sh << EOF
+#!/bin/sh
+cp "\$RENEWED_LINEAGE/fullchain.pem" /etc/nginx/private/cert.pem
+cp "\$RENEWED_LINEAGE/privkey.pem" /etc/nginx/private/key.pem
+EOF
+
+cat > /etc/periodic/weekly/reload-nginx.sh << EOF
```
Comment (henokgetachew) [inline nginx/ssl-install.sh:121]: I get the idea. Question: Is it possible to use certbot's ` --deploy-hook` instead?

**Review by henokgetachew — COMMENTED** [review]
I like this! I have just sent you one question in line.

**Review by henokgetachew — APPROVED** [review]
LGTM

File: nginx/ssl-install.sh (hunk 1 above)
Comment (henokgetachew) [reply nginx/ssl-install.sh:121]: Great! Yeah that makes sense.

**Pull request merged**